### DISCUSSION TITLE
Where are the exact directories of the modules for addon development?

### DISCUSSION THREAD

**Discussion by NelloHo** [nvaccess/nvda#13607] [Q&A]
There are five modules that we often import them to our python addon scripts:
appModules
brailleDisplayDrivers
globalPlugins
synthDrivers
visionEnhancementProviders
Does anyone know where their exact directories are ? I want to view the code written in these modules.

**Comment by CyrilleB79** [comment]
I hope you have found the answer in the meantime. Anyway if it's useful for someone else:

Open the Python console (NVDA+control+Z). And type the following commands:
```
>>> import appModules, brailleDisplayDrivers, globalPlugins, synthDrivers, visionEnhancementProviders
>>> appModules
<module 'appModules' from 'C:\\Program Files (x86)\\NVDA\\library.zip\\appModules\\__init__.pyc'>
>>> brailleDisplayDrivers
<module 'brailleDisplayDrivers' from 'C:\\Program Files (x86)\\NVDA\\library.zip\\brailleDisplayDrivers\\__init__.pyc'>
>>> globalPlugins
<module 'globalPlugins' from 'C:\\Program Files (x86)\\NVDA\\library.zip\\globalPlugins\\__init__.pyc'>
>>> synthDrivers
<module 'synthDrivers' from 'C:\\Program Files (x86)\\NVDA\\library.zip\\synthDrivers\\__init__.pyc'>
>>> visionEnhancementProviders
<module 'visionEnhancementProviders' from 'C:\\Program Files (x86)\\NVDA\\library.zip\\visionEnhancementProviders\\__init__.pyc'>
```

If you clone NVDA's repo in `path/to/the/cloned/repo/of/nvda`, you will have the files in:
`path/to/the/cloned/repo/of/nvda/source`. For example, `appModules` module is in `path/to/the/cloned/repo/of/nvda/source/appModules/__init__.py`.

**Comment by CyrilleB79** [comment]
Note however that we often import corresponding classes rather than the module.

For example:
`from appModuleHandler import AppModule`
We import the `AppModule` class, not the `appModules` folder; beware to the initial capital and the missing final "s".
In this case, in the console:
```
>>> from appModuleHandler import AppModule
>>> AppModule
<class 'appModuleHandler.AppModule'>
```

The class is defined in 
`path\to\the\cloned\repo/of\nvda\source\appModuleHandler.py`.
